x-pack/platform/plugins/private/reporting/server/services/scheduled_reports/scheduled_reports_service.ts

```diff
@@ -299,7 +302,187 @@ export class ScheduledReportsService {
     }
   }
 
-  private getUsername(user: ReportingUser): string | Boolean {
+  public async bulkDelete({
+    ids,
+    user,
+  }: {
+    ids: string[];
+    user: ReportingUser;
+  }): Promise<BulkDeleteResult> {
+    try {
+      const username = this.getUsername(user);
+
+      const bulkGetResult = await this.savedObjectsClient.bulkGet<ScheduledReportType>(
+        ids.map((id) => ({ id, type: SCHEDULED_REPORT_SAVED_OBJECT_TYPE }))
+      );
+
+      const {
+        readErrors = [],
+        unauthorizedSavedObjects = [],
+        authorizedSavedObjects = [],
+      } = groupBy(bulkGetResult.saved_objects, (so) =>
+        so.error
+          ? 'readErrors'
```

Comment:
I think it is better if we group the errors by their status code (`error.statusCode`). To make it simpler, you can even partition (from lodash) the SOs and have one array with SOs without errors and one array with errors. Then you can process the errors by status codes. Example:

```
 const [validReports, soBulkGetErrors] = partition(
      bulkGetResult.saved_objects,
      (so) => so.error === undefined
    )

// then process errors by status codes or whatever is needed.
```

Reply:
Actually there is no logic related to error status codes, and the authorization check is performed on successfully retrieved schedule objects (to check the user authorship): `groupBy` was useful to do everything in one go here. I changed it with two `partition`s.

nit: I didn't like the original `for` loops either - and we probably don't have strong performance needs here - but with this flow we're looping on the same data many more times 😬